core/slice.go

```diff
@@ -289,6 +291,207 @@ func (sl *Slice) Append(header *types.WorkObject, domTerminus common.Hash, domOr
 		}
 	}
 
+	if nodeCtx == common.PRIME_CTX {
+		// Until the controller has kicked in, dont update any of the fields,
+		// and make sure that the fields are unchanged from the default value
+		if header.NumberU64(common.PRIME_CTX) <= params.ControllerKickInBlock {
```

Comment:
If we are sure that this is always the same as PrimeTerminusNumber, this is fine, but I used wo.PrimeTerminusNumber in all my checks

Reply:
it happens before the prime terminus number changes